### PR TITLE
supplemental: refactor misaligned leading asterisks in javadoc comments

### PR DESCRIPTION
**Pull request by Zopsss**
Supplemental for #6723 

Refactored misaligned leading asterisks in javadoc comments.

### CODE REVIEW

**Review by rnveach — APPROVED** [review]

**Comment by Zopsss** [comment]
@nrmancuso can you please merge this PR? Because [LeadingAsteriskAlign](https://github.com/checkstyle/checkstyle/pull/14643) PR will be blocked until this PR gets merged.

**Review by romani — APPROVED** [review]
ok to merge

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java:154]: Why does this check care about extra *?

Comment (Zopsss) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java:154]: The check detected this in the older implementation due to some bug, now the current implementation has fixed that bug. I thought it would be okay to remove this leading asterisk since it is unnecessary to have it

Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java:154]: Ok, please make sure to add this to our test cases in your new check PR

**Pull request merged**
